libs/smiles/test/test_standardize_smiles.py

```diff
@@ -0,0 +1,63 @@
+import os
+import pandas as pd
+import tempfile
+from src.smiles.standardize_smiles import StandardizeMolecule
+import pytest
+
+from pathlib import Path
+
+test_data_dir = Path(__file__).resolve().parent / "test_data"
+
+
+@pytest.mark.parametrize(
+    "method",
+    [
+        "jump_canonical",
+        "jump_alternate_1",
+    ],
+)
+def test_standardize_molecule(method):
+    input_file = str(
+        (
+            test_data_dir
+            / "smiles_data"
+            / "JUMP-Target-2_compound_metadata_trimmed_input.tsv"
+        ).resolve()
+    )
+
+    expected_output_file = str(
+        (
+            test_data_dir
+            / "smiles_data"
+            / f"JUMP-Target-2_compound_metadata_trimmed_output_{method}.csv"
+        ).resolve()
+    )
+
+    tmpdir = tempfile.gettempdir()
+
+    temp_output_file = os.path.join(tmpdir, "test.csv")
+
+    # Create an instance of StandardizeMolecule with the desired parameters
+    standardizer = StandardizeMolecule(
+        input=input_file,
+        output=temp_output_file,
+        num_cpu=4,
+        augment=True,
+        method="jump_alternate_1",
+        random_seed=42,
+    )
+
+    # Run the standardization process
+    standardizer.run()
```

Comment:
After homogeneising the interface (so this always returns a dataframe). Then, checking the individual columns is important to find which process went wrong.

```
result = standardizer.run()
for col in ("SMILES", "InChi", "InChiKey"):
    assert result[f"{col}_standardized"] == validation[f"{col}_standardized"]
```
Finally, add equivalent tests to _standardize_structure. 

The goal of all this is to have fine-grained understanding of each processing branch. The current testing method only checks for dataframe equality, and that contains a lot of noise and additional formatting.

Reply:
Great! tests are more fine-grained now. Everything is still tested together but at least the assertions test for failures in a more fine-grained manner